### DISCUSSION TITLE
Get RSI to match Trading View

### DISCUSSION THREAD

**Discussion by duthiega** [DaveSkender/Stock.Indicators#1145] [Help and support]
Could someone help me get the same RSI seen on TradingView using Binance? 

https://www.tradingview.com/symbols/BTCUSDT/technicals/?exchange=BINANCE 

/api/v3/klines?endTime={endTime}&limit={limit}&interval={timePeriod}&symbol=BTCUSDT

I not 100% sure that my understanding is correct as the documentation says "You must have at least N+100 periods of quotes to cover the convergence periods"

I have tried passing 140 as the limit above and used them to call 
`var rsiResult = candlesticks.GetRsi(14);`  
on my candlesticks, which implement IQuote 

I've tried
`rsiResult.First(e=>e.Rsi != null).Rsi`
`rsiResult.Last(e=>e.Rsi != null).Rsi`

I've also tried other things like adding utility methods, but have varied results. Sometimes its within 1%, sometimes its up to 15% out. Any help on this would be greatly appreciated 

Condense()
Find(lookupDate)
RemoveWarmupPeriods()

Maybe its just the fact i dont fully understand why i need to have a 10 times more data points

**Comment by DaveSkender** [comment]
RSI and many other indicators use a [convergence](https://en.wikipedia.org/wiki/Convergence_of_random_variables) type of smoothing average.  As such, initial values are essentially a smart guess are often less accurate, as you're depicting.  Over time they converge to be consistent values.  To overcome, use _more_ historical quotes that precede your evaluation period.  More info:

- #688 
- #801

**Reply by duthiega** [reply]
Thanks for your quick reply and after running the following code I can now see the convergence and that adding the  `.RemoveWarmupPeriods() `removes `(rsiPeriod * 10) `as defined in the docs

```
var baseAsset = "BTC";
var quoteAsset = "USDT";
var timePeriod = "1m";
var rsiPeriod = 14;
var binanceService = host.Services.GetService<ICryptoExchange>();
var candlesticks = await binanceService.GetKlinesAsync(baseAsset, quoteAsset, timePeriod, rsiPeriod * 20 + rsiPeriod, DateTime.Now);

for(int i = 1; i < 20; i++)
{
    var quotes = candlesticks.OrderByDescending(c=>c.CloseTime).Take(rsiPeriod * i + rsiPeriod); 
    var result = quotes.GetRsi(rsiPeriod);
    Console.WriteLine($"Count : {rsiPeriod * i + rsiPeriod} \t RSI : {result.Last(e => e.Rsi != null).Rsi}  \t Date : {result.Last(e => e.Rsi != null).Date}");
}

> 
Count : 28       RSI : 33.609958771884266        Date : 09/01/2024 10:57:59
Count : 42       RSI : 36.18886161713568         Date : 09/01/2024 10:57:59
Count : 56       RSI : 37.166274744539905        Date : 09/01/2024 10:57:59
Count : 70       RSI : 36.840183294934505        Date : 09/01/2024 10:57:59
Count : 84       RSI : 36.65081273847727         Date : 09/01/2024 10:57:59
Count : 98       RSI : 36.66459366302223         Date : 09/01/2024 10:57:59
Count : 112      RSI : 36.64550123867035         Date : 09/01/2024 10:57:59
Count : 126      RSI : 36.64792797763708         Date : 09/01/2024 10:57:59
Count : 140      RSI : 36.64524849913927         Date : 09/01/2024 10:57:59
Count : 154      RSI : 36.64523357959248         Date : 09/01/2024 10:57:59
Count : 168      RSI : 36.64568503376472         Date : 09/01/2024 10:57:59
Count : 182      RSI : 36.645568418046764        Date : 09/01/2024 10:57:59
Count : 196      RSI : 36.64559276217767         Date : 09/01/2024 10:57:59
Count : 210      RSI : 36.64557924947504         Date : 09/01/2024 10:57:59
Count : 224      RSI : 36.64558673482039         Date : 09/01/2024 10:57:59
Count : 238      RSI : 36.645585272878726        Date : 09/01/2024 10:57:59
Count : 252      RSI : 36.64558592713405         Date : 09/01/2024 10:57:59
Count : 266      RSI : 36.64558562415006         Date : 09/01/2024 10:57:59
Count : 280      RSI : 36.6455856168194          Date : 09/01/2024 10:57:59
```
However, I am still not able to match the RSI with the value displayed on trading view.   Once again, any help to try and match these would be appreciated 
https://www.tradingview.com/symbols/BTCUSDT/technicals/?exchange=BINANCE

**Reply by DaveSkender** [reply]
Compare the underlying quotes.  You and TradingView may be working from different base price data.  Crypto exchange data is especially wild across providers as they may aggregate differently and sometimes exclude weekends.

**Reply by duthiega** [reply]
Thanks again @DaveSkender.  I suppose its all relative. And, if I'm going to be running any ML against outputs of various technical indicators, consistency will be the main factor.

**Reply by 030Studios** [reply]
its possible to get the trading view RSI. You just need 1000 Bars. For example in 1 min intervall. Then just calculate with GetRsi.
